go/controller/api/v1alpha1/deploymentsettings_types.go

```diff
@@ -79,56 +135,123 @@ func (r *HTTPConnection) Attributes(ctx context.Context, c client.Client, namesp
-//+kubebuilder:subresource:status
+// AISettings holds the configuration for LLM provider clients.
+type AISettings struct {
+	// Enabled defines whether to enable the AI integration or not.
+	//
+	// +kubebuilder:default=false
+	// +kubebuilder:validation:Optional
+	Enabled *bool `json:"enabled,omitempty"`
 
-// DeploymentSettings is the Schema for the deploymentsettings API
-type DeploymentSettings struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Provider defines which of the supported LLM providers should be used.
+	//
+	// +kubebuilder:validation:Enum=OPENAI;ANTHROPIC
+	// +kubebuilder:default=OPENAI
+	// +kubebuilder:validation:Optional
+	Provider *console.AiProvider `json:"provider,omitempty"`
 
-	Spec   DeploymentSettingsSpec `json:"spec,omitempty"`
-	Status Status                 `json:"status,omitempty"`
+	// OpenAI holds the OpenAI provider configuration.
+	//
+	// +kubebuilder:validation:Optional
+	OpenAI *AIProviderSettings `json:"openAI,omitempty"`
+
+	// Anthropic holds the Anthropic provider configuration.
+	//
+	// +kubebuilder:validation:Optional
+	Anthropic *AIProviderSettings `json:"anthropic,omitempty"`
 }
 
-//+kubebuilder:object:root=true
+func (in *AISettings) Attributes(ctx context.Context, c client.Client, namespace string) (*console.AiSettingsAttributes, error) {
+	attr := &console.AiSettingsAttributes{
+		Enabled:  in.Enabled,
+		Provider: in.Provider,
+	}
 
-// DeploymentSettingsList contains a list of DeploymentSettings
-type DeploymentSettingsList struct {
-	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []DeploymentSettings `json:"items"`
+	switch *in.Provider {
+	case console.AiProviderOpenai:
+		token, err := in.OpenAI.Token(ctx, c, namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		attr.Openai = &console.OpenaiSettingsAttributes{
+			AccessToken: &token,
+			Model:       &in.OpenAI.Model,
+		}
+	case console.AiProviderAnthropic:
+		token, err := in.Anthropic.Token(ctx, c, namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		attr.Anthropic = &console.AnthropicSettingsAttributes{
+			AccessToken: &token,
+			Model:       &in.OpenAI.Model,
+		}
+	}
+
+	return attr, nil
 }
 
-func init() {
-	SchemeBuilder.Register(&DeploymentSettings{}, &DeploymentSettingsList{})
+type AIProviderSettings struct {
+	// Model is the LLM model name to use.
+	//
+	// +kubebuilder:validation:Required
+	Model string `json:"model"`
+
+	// TokenSecretRef is a reference to the local secret holding the token to access
+	// the configured AI provider.
+	//
+	// +kubebuilder:validation:Required
+	TokenSecretRef corev1.SecretKeySelector `json:"tokenSecretRef"`
 }
 
-func (p *DeploymentSettings) SetCondition(condition metav1.Condition) {
-	meta.SetStatusCondition(&p.Status.Conditions, condition)
+func (in *AIProviderSettings) Token(ctx context.Context, c client.Client, namespace string) (string, error) {
```

Comment:
don't we have some reusable code to grab these?

Reply:
I couldn't find any function that took care of that so I have extracted this logic to a reusable one.